course/page/upload.py

```diff
@@ -294,7 +308,8 @@ def normalized_bytes_answer(self, page_context, page_data, answer_data):
 
         subm_data, subm_mime = self.get_content_from_answer_data(answer_data)
 
-        from mimetypes import guess_extension
+        from mimetypes import add_type, guess_extension
+        add_type('application/x-ipynb+json', '.ipynb')
```

Comment:
Maybe move this to the module level? There are other uses of `guess_extension` that might also benefit.

Reply:
done.  it's at the top top